File: 136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        // Initialize a variable to 0. It will hold the XOR sum.
        int singleElement = 0;
        
        // Iterate through each number in the input array.
        for (int num : nums) {
            // Apply the XOR operation between the current number
            // and the accumulated XOR sum.
            singleElement ^= num;
        }
        
        // After the loop, the duplicates have canceled out,
        // leaving only the single number.
        return singleElement;
    }
}

File: 136-single-number/single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # Create a list of numbers from 0 to 9
        # Iterate throught the list and compare with the nums list
        # If count of the particular number is greater than 1 'continue'
        # Else return the nums element that has 1 frequency

        single_element = 0
        for i in nums:
            single_element ^= i
        return single_element 
        
        